src/v/kafka/server/group.cc

```diff
@@ -1474,6 +1474,14 @@ void group::remove_member(member_ptr member) {
     }
 }
 
+void group::remove_full_members() {
+    while (!_members.empty()) {
+        auto member = _members.begin()->second;
+        member->expire_timer().cancel();
+        remove_member(member);
+    }
+}
```

Comment:
what about pending members?  

unrelated to this PR.. I have a hard time to trying to reason about how member state changes are persisted 🤔

Reply:
> what about pending members?

They don't seem to block group deletion

> unrelated to this PR.. I have a hard time to trying to reason about how member state changes are persisted 🤔

Yeah the one where we ignore replication result scares me the most.

Maybe instead of removing members one-by-one and waiting for the rebalance we should just force-delete the group and replicate a tombstone for it regardless its members status. I should have asked about it in the joint discussion yesterday.